reloc.ml

```diff
@@ -269,12 +269,16 @@ let find_file =
     with Not_found ->
       try Hashtbl.find memo (k ^ ".lib")
       with Not_found ->
-        let fn =
+        let fns =
           if String.length fn > 2 && String.sub fn 0 2 = "-l" then
-            "lib" ^ (String.sub fn 2 (String.length fn - 2))
-          else fn in
+            let base = String.sub fn 2 (String.length fn - 2) in
+            if !toolchain = `MSVC || !toolchain = `MSVC64 then
+              ["lib" ^ base; base]
+            else
+              ["lib" ^ base]
+          else [fn] in
         let r =
-          match find_file fn with
+          match List.fold_left (function None -> find_file | acc -> fun _ -> acc) None fns with
```

Comment:
This is not very readable.  Could you introduce some helper function such as `val map_find: ('a -> 'b option) -> 'a list -> 'b option` (implemented as a recursive function)?

Otherwise, the PR looks good to me.

Reply:
Yes indeed, that's a bit hideous - sorry! Factored out...